ansible_collections/arista/avd/tests/unit/filters/test_ipaddr.py

```diff
@@ -0,0 +1,151 @@
+                             ['192.24.2.0', '::', '192.168.31.255', 'fe80::ff', '192.168.32.0', '192.168.1.254', '192.168.1.33', '192.168.1.33']
+                             ]
+IPV4_ADDRESSES_VALIDATE_EXPECTED = ['192.24.2.1', '192.168.32.0/24', '192.168.32.1/24', '192.168.1.255/24', '192.168.1.34/28',
+                                    '192.168.1.34/25']
+
+
+def get_by_method(IP_ADDRESS_INPUT_LIST=None, method='validate'):
+    if IP_ADDRESS_INPUT_LIST is None:
+        return []
+    resp_list = []
+    for IP_ADDRESS in IP_ADDRESS_INPUT_LIST:
+        if isinstance(IP_ADDRESS, bool) or IP_ADDRESS in ['', [], None]:
+            continue
+        if res := getattr(f, method)(IP_ADDRESS):
+            if not isinstance(res, int):
+                res = str(res)
+            resp_list.append(res)
+    return resp_list
+
+
+def ipv4_ipv6_assert(INPUT_RESP, EXPECTED_OUTPUT):
+    # f.ipv4(output) --> filters IPv4 output by leveraging validate method
+    assert f.ipv4(INPUT_RESP) == f.ipv4(EXPECTED_OUTPUT)
+    assert f.ipv6(INPUT_RESP) == f.ipv6(EXPECTED_OUTPUT)
+
+
+class TestIpAddrFilter():
+    @pytest.mark.parametrize("VALIDATE_INPUT,VALIDATE_EXPECTED", VALIDATE_LIST_INPUT_EXPECTED)
+    def test_validate_function(self, VALIDATE_INPUT, VALIDATE_EXPECTED):
+        resp = f.validate(VALIDATE_INPUT)
+        if resp:
+            resp = str(resp)
+        assert resp == VALIDATE_EXPECTED
+
+    @pytest.mark.parametrize("IP_ADDRESS", IP_ADDRESSES_INPUT)
+    def test_ipaddr_validate_item(self, IP_ADDRESS):
+        resp = f.ipaddr(IP_ADDRESS)
+        expected_ip = f.validate(IP_ADDRESS)
+        if type(IP_ADDRESS) in (str, int) and expected_ip:
+            assert resp == str(expected_ip)
+        else:
+            assert resp is None
+
+    @pytest.mark.parametrize("IP_ADDRESS_LIST_INPUT,IP_ADDRESS_LIST_EXPECTED", [(IP_ADDRESSES_INPUT, IP_ADDRESSES_VALIDATE_EXPECTED)])
+    def test_ipaddr_validate_list(self, IP_ADDRESS_LIST_INPUT, IP_ADDRESS_LIST_EXPECTED):
+        resp_list = f.ipaddr(IP_ADDRESS_LIST_INPUT)
+        assert resp_list == IP_ADDRESS_LIST_EXPECTED
+        assert get_by_method(IP_ADDRESS_LIST_INPUT, 'validate') == IP_ADDRESS_LIST_EXPECTED
+        ipv4_ipv6_assert(resp_list, IP_ADDRESS_LIST_EXPECTED)
+
+    @pytest.mark.parametrize("IP_ADDRESS_LIST_INPUT,IP_ADDRESS_LIST_EXPECTED", [(IP_ADDRESSES_INPUT, IP_ADDRESSES_NET_EXPECTED)])
+    def test_ipaddr_network_list(self, IP_ADDRESS_LIST_INPUT, IP_ADDRESS_LIST_EXPECTED):
+        resp_list = f.ipaddr(IP_ADDRESS_LIST_INPUT, method='net')
+        assert resp_list == IP_ADDRESS_LIST_EXPECTED
+        assert get_by_method(IP_ADDRESS_LIST_INPUT, 'net') == IP_ADDRESS_LIST_EXPECTED
+        ipv4_ipv6_assert(resp_list, IP_ADDRESS_LIST_EXPECTED)
+
+    @pytest.mark.parametrize("IP_ADDRESS_LIST_INPUT,IP_ADDRESS_LIST_EXPECTED", [(IP_ADDRESSES_INPUT, IP_ADDRESSES_SUBNETWORK_EXPECTED)])
+    def test_ipaddr_subnetwork_list(self, IP_ADDRESS_LIST_INPUT, IP_ADDRESS_LIST_EXPECTED):
```

Comment:
It is not apparent what we are testing in each scenario. It would be great with a docstring for each test function.